test/NuGet.Core.Tests/NuGet.Commands.Test/MSBuildRestoreUtilityTests.cs

```diff
@@ -3399,6 +3400,27 @@ public void MSBuildRestoreUtility_AddPackageDownloads_InvalidVersionRange(string
             var exception = Assert.Throws<ArgumentException>(() => MSBuildRestoreUtility.AddPackageDownloads(spec, msbuildItems));
         }
 
+        [Fact]
+        public void MSBuildRestoreUtility_AddPackageDownloads_NoVersion_ThrowsException()
+        {
+            // Arrange
+            PackageSpec spec = MSBuildRestoreUtility.GetPackageSpec(new[] { CreateItems(new Dictionary<string, string>()) });
+            var packageX = new Mock<IMSBuildItem>();
+            const string packageId = "x";
+            packageX.Setup(p => p.GetProperty("Type")).Returns("DownloadDependency");
+            packageX.Setup(p => p.GetProperty("Id")).Returns(packageId);
+
+            var msbuildItems = new[]
```

Comment:
I'd prefer to see the type somehow: either an explicit type instead of var, or an array type declaration.